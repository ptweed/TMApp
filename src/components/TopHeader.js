import React, { Component } from 'react';
import { Platform, StyleSheet, View, Text, SafeAreaView } from 'react-native'
import { FontAwesome5 } from '@expo/vector-icons'
import Captions from '../utils/Captions'
import IconButton from './IconButton'
import SimpleButton from './SimpleButton'

export default class TopHeader extends Component {
  render() {

    let localStyle={paddingLeft: 15, color: 'black' }
    if (!this.props.isConnected) {
      localStyle = {paddingLeft: 15, color: 'grey' }
    }
    return (
      <SafeAreaView style={styles.headerContainer}>
        <View style={styles.leftButtonContainer}>
          <IconButton iconColor='black'  
          iconSize={20} iconName='plus' onPress={() => { this.props.navigation.navigate('TaskCreate')}} 
          disabled={!this.props.isConnected} />
        </View>
        <View style={{flex: 2, alignItems: 'center', flexDirection: 'column', marginBottom: 5}}>
          <Text style={{fontSize: 20}}>{Captions.appName}</Text>
          <FontAwesome5 name="tasks" size={20} />
        </View>
        <View style={styles.rightButtonContainer}>
          <SimpleButton title={Captions.logout} textColor='black' textSize={18} onPress={this.props.logout} />
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  headerContainer: {
    borderBottomWidth: 1,
    flexDirection:'row',
    alignItems:'center',
    borderColor: "black",
    backgroundColor: "rgb(255,255,255)",
    marginTop: 10,
    paddingBottom: 5, 
  },
  leftButtonContainer: {
    flex: 1, 
    marginLeft:20, 
    alignItems: 'flex-start',
    ...Platform.select({
      ios: {
          },
      android: {
        marginTop: 15,
          },
      default: {
        // other platforms, web for example
      }
    })
  },
  rightButtonContainer: {
    flex: 1, 
    marginRight:20, 
    alignItems: 'flex-end',
    ...Platform.select({
      ios: {
          },
      android: {
        marginTop: 15,
          },
      default: {
        // other platforms, web for example
      }
    })
  },
})