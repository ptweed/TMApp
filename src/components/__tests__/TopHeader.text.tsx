import React from 'react'
import { render } from 'react-native-testing-library'
import TopHeader from '../TopHeader'

jest.mock('@expo/vector-icons', () => ({
    FontAwesome5: ''
}))
  
describe('TopHeader Component', () => {
  it('TopHeader renders correctly', () => {
    const onPressEvent = jest.fn();
    const topHeader = render(<TopHeader isConneced={true} onPress={onPressEvent} />).toJSON();
    expect(topHeader).toMatchSnapshot();
  });

});