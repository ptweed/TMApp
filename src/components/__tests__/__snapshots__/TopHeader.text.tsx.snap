// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopHeader Component TopHeader renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(255,255,255)",
      "borderBottomWidth": 1,
      "borderColor": "black",
      "flexDirection": "row",
      "marginTop": 10,
      "paddingBottom": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "marginLeft": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="topLevelButtonContainer"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <
          color="gray"
          name="plus"
          size={20}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "flexDirection": "column",
        "marginBottom": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Task Master
    </Text>
    <
      name="tasks"
      size={20}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "marginRight": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="topLevelButtonContainer"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 18,
            }
          }
        >
          Logout
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
